src/cli/util.js

```diff
@@ -401,51 +402,106 @@ function createIgnorerFromContextOrDie(context) {
   }
 }
 
-function eachFilename(context, patterns, callback) {
-  // workaround for fast-glob on Windows ref:
-  // https://github.com/mrmlnc/fast-glob#how-to-write-patterns-on-windows
-  if (path.sep === "\\") {
-    patterns = patterns.map(path => path.replace(/\\/g, "/"));
+/**
+ * Get stats of a given path.
+ * @param {string} filePath The path to target file.
+ * @returns {fs.Stats|undefined} The stats.
+ * @private
+ */
+function statSafeSync(filePath) {
+  try {
+    return fs.statSync(filePath);
+  } catch (error) {
+    /* istanbul ignore next */
+    if (error.code !== "ENOENT") {
+      throw error;
+    }
   }
+}
 
-  // The '!./' globs are due to https://github.com/prettier/prettier/issues/2110
-  const ignoreNodeModules = context.argv["with-node-modules"] !== true;
-  if (ignoreNodeModules) {
-    patterns = patterns.concat(["!**/node_modules/**", "!./node_modules/**"]);
+// `dot pattern` and `expand directories` support need handle differently
+// for backward compatibility reason temporary remove `.` and set `expandDirectories=false`
+// see https://github.com/prettier/prettier/pull/6639#issuecomment-548949954
```

Comment:
This comment is still confusing to me. Yes, we're going to expand directories sometime in the future. But what does this have to do with backward compatibility?

Reply:
Try read little more from https://github.com/prettier/prettier/pull/6639#issuecomment-548634927 , if this still  
 confusing you, I'll try to explain.

Reply:
I think I understand it, it's just that I wouldn't call it backward compatibility. Never mind.